### PR TITLE
migrate to vendor

### PR DESCRIPTION
**Pull request by unclejack**
This PR migrates ofnet to vendor.

This PR makes the following changes:
- bump vishvananda/netlink to the latest version
- bump vishvananda/netns to the latest version
- bumps libovsdb to the latest version
- ~switches to the centos/73 box to use Go 1.7.5~ delayed for a future PR
- fixes some minor gofmt and go vet issues to pass checks

Most of these changes were made in one commit because godep was causing some issues with vendor.

### CODE REVIEW

**Comment by dseevr** [comment]
Changes look fine but the build is failing...

**Comment by unclejack** [comment]
The failures don't seem to be related to the libovsdb, netlink and netns. Reverting those doesn't fix the issue.

**Comment by unclejack** [comment]
@jojimt: Could you take a look at the logs when you have the time, please? I'm not sure why it's failing right now.

**Comment by unclejack** [comment]
I've reverted the change which was switching to `contiv/centos73`. This fixes the failure we were seeing in the tests.

I've opened an issue to track the TestPolicyAddDelete failure: #87.

This PR will allow us to bump the other deps (libovsdb, libOpenflow).

This PR is ready for review.
@dseevr @jojimt PTAL

File: Makefile [hunk 1]
```diff
@@ -20,9 +20,9 @@ test: build
...
 	./checks "./*.go ./libpkt/ ./ofctrl/ ./ovsdbDriver/ ./ovsSwitch/ ./pqueue/ ./rpcHub/"
 	go get github.com/tools/godep
-	godep go install ./ ./ofctrl
+	go install ./ ./ofctrl
 
 host-test:
-	sudo -E PATH=$(PATH) /opt/gopath/bin/godep go test -v ./
-	sudo -E PATH=$(PATH) /opt/gopath/bin/godep go test -v ./ofctrl
-	sudo -E PATH=$(PATH) /opt/gopath/bin/godep go test -v ./pqueue
+	PATH=${PATH} sudo -E /usr/local/go/bin/go test -v ./
```
Comment (dseevr) [inline Makefile]: If the box isn't being updated, do we still need the `host-test` path changes in `Makefile`?  IIRC the old box had Go installed under the `/opt` path

Comment (unclejack) [inline Makefile]: @dseevr: This makes it easy for us to migrate away from godep. Even though we're not upgrading to Go 1.7, we're on Go 1.6 and we can get rid of godep in some places. The new box has Go in the same place, but it's easier to not rely on the box's gopath.

Comment (dseevr) [inline Makefile]: Cool 👍

**Comment by abhinandanpb** [comment]
the last time we tried moving ofnet to vendoring (go1.6) there was dependency issue when ofnet was vendored into netplugin. Did you get a chance to test out netplugin with the vendoring ?

**Comment by unclejack** [comment]
@abhinandanpb: I'll try this out and post a reply. The Godeps workspace was broken without this PR. I can imagine that's what was breaking it before.

**Comment by abhinandanpb** [comment]
what was broken ? We have been using godeps forever with bunch of dependency brought in and out. We havent had any issues with godeps in ofnet.

**Comment by unclejack** [comment]
@abhinandanpb: How are you using godeps to update dependencies in ofnet? I get errors.

**Comment by unclejack** [comment]
@abhinandanpb: Please let me know if there's anything else you'd like me to do or look into.

**Comment by unclejack** [comment]
This PR is rebased on top of master.

I haven't changed the version of the Vagrant box. We have to address some failures with Go 1.7 in a separate PR.

The libOpenflow library is also not upgraded or modified in any way in this PR. That's to be done in a separate PR.

@DivyaVavili @gkvijay @rhim: We need this PR. PTAL

**Comment by DivyaVavili** [comment]
Also, what is the recommended way of moving to vendor when it is used in other repositories? Will it even make a difference?

**Comment by dseevr** [comment]
@DivyaVavili it doesn't make a difference as to how the code is used under `vendor`... run this in netplugin and you'll see we still have one dependency using the legacy `_workspace` dir with our modern vendoring: `find vendor -type d -name _workspace`

**Comment by unclejack** [comment]
@DivyaVavili: There's nothing to check with netplugin in this case. ofnet itself is functioning properly and it's brought in with all of its dependencies. The difference is that we won't be running into errors any more when we try to bump dependencies.

We need to make progress and improve the code, but this is a major roadblock.

**Comment by dseevr** [comment]
The full netplugin CI suite will be run when netplugin's Godeps are updated to use the new version of this code.

We _need_ to switch to vendor.  It's been the ordained way to manage dependencies in Go projects for almost 1.5 years, and our code no longer compiles with any version of Go which does not support vendoring.

We should merge this ASAP so that we can open the PR to update this dependency in netplugin itself

**Comment by unclejack** [comment]
I've tested this with netplugin. ofnet was bumped properly. There's some work left to be done when we bump this in netplugin itself. This work which needs to be done isn't something which should block this PR, though. It should be safe to proceed at this point.

**Comment by DivyaVavili** [comment]
Thanks for checking this. LGTM with respect to this change. But when we are including it into contiv/netplugin, let's check it rigorously before merging in for the next major release so that we don't run into any dependency issues.

**Pull request merged**